innovation.game.php

```diff
@@ -10486,6 +10505,58 @@ function stInterInteractionStep() {
                     }
                     break;
                 
+				// id 123, Artifacts age 1: Ark of the covenant
+                case "123N1A":
+					$players = self::loadPlayersBasicInfos();
+					
+					if ($n > 0)
+					{ // Unsaid rule: the player must have at least one card to show from his hand, else, the effect can't continue
+                        $color = self::getGameStateValue('color_last_selected');
+                            
+                        foreach($players as $all_player_id => $player) 
+						{
+                            // TODO : is there a better way to get all top cards?
+							$top_blue_card = self::getTopCardOnBoard($all_player_id, 0);
+							$top_red_card = self::getTopCardOnBoard($all_player_id, 1);
+							$top_green_card = self::getTopCardOnBoard($all_player_id, 2);
+							$top_yellow_card = self::getTopCardOnBoard($all_player_id, 3);
+							$top_purple_card = self::getTopCardOnBoard($all_player_id, 4);
+							
+							// TODO : if there is an artifact card on top of any pile, then prevent the rest of the dogma.
+							if (($top_blue_card == null || $top_blue_card['id'] < 110) &&
+								($top_red_card == null || $top_red_card['id'] < 110) &&
+								($top_green_card == null || $top_green_card['id'] < 110) &&
+								($top_yellow_card == null || $top_yellow_card['id'] < 110) &&
+								($top_purple_card == null || $top_purple_card['id'] < 110))
+							{
+								// No artifacts as a top card, transfer all cards to the original player's
+								// scorepile
+								$top_card = self::getTopCardOnBoard($all_player_id, $color);
+								while ($top_card !== null)
+								{	
+									// "Transfer all cards of the same color from the boards of all players
+									// with no top artifacts to your score pile."
+									// This is a "transfer" so monument cannot be achieved with this dogma.
```

Comment:
Let's just omit this comment to keep things shorter.

Reply:
sounds good